platform/iphone/gl_view.mm

```diff
@@ -379,369 +222,106 @@ - (void)destroyFramebuffer {
 }
 
 - (void)startAnimation {
-	if (active)
-		return;
-	active = TRUE;
-	printf("start animation!\n");
-	if (useCADisplayLink) {
+	// Already active?
+	if (self.isActive) return;
+
+	self.active = YES;
 
-		// Approximate frame rate
-		// assumes device refreshes at 60 fps
-		int frameInterval = (int)floor(animationInterval * 60.0f);
+	printf("start animation!\n");
 
-		displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(drawView)];
-		[displayLink setFrameInterval:frameInterval];
+	if (self.useCADisplayLink) {
+
+		// Create a display link and try and grab the fatest refresh rate
+		_displayLink = [CADisplayLink displayLinkWithTarget:self
+												   selector:@selector(drawView)];
+		if ([_displayLink respondsToSelector:@selector(preferredFramesPerSecond)]) {
+			_displayLink.preferredFramesPerSecond = 0;
+			// We could potentially check the display link's maximumFPS here.
+			// If it supports 120 we could force the preferredFPS to that.
+			// Currently that the's only way to drive a game at 120Hz
+		} else {
+			// Approximate frame rate: assumes device refreshes at 60 fps.
+			// Note that newer iOS devices are 120Hz screens
+			_displayLink.frameInterval = 1;
```

Comment:
I have seen these two properties trip people up probably more than any other Apple API.

https://developer.apple.com/documentation/quartzcore/cadisplaylink/1648421-preferredframespersecond?language=objc
https://developer.apple.com/documentation/quartzcore/cadisplaylink/1621231-frameinterval?language=objc

We _really do_ want to pass `0` and `1` as the requested FPS.

Reply:
The previous code raised this warning (Travis CI), does this change fix it?
```
platform/iphone/gl_view.mm:399:16: warning: 'setFrameInterval:' is deprecated: first deprecated in iOS 10.0 - use preferredFramesPerSecond [-Wdeprecated-declarations,29]
                [displayLink setFrameInterval:frameInterval];
                              ^
/Applications/Xcode-9.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CADisplayLink.h:70:32: note: property 'frameInterval' is declared deprecated here [2]
 @property(nonatomic) NSInteger frameInterval
                                ^
```